### ISSUE TITLE
Selecting SQL Server column of type `date` converts to US style and adds colon before ‘AM’

### ISSUE THREAD

**Issue by kokoshneta** [laravel/framework#41799]
- Laravel Version: 9.3.1 (also present in 8.X before I upgraded)
- PHP Version: 8.1.3 running on macOS 10.15.7 and 12.2.1
- Database Driver & Version: SQL Server 2019 (15.00.2080) on Windows Server 2019 Standard (1809, build 17763.2686)

### Description:

Whenever any table in an SQL Server database has a column of type `date`, fetching that column from within Laravel consistently breaks the value in the exact same way. This happens both in Eloquent models and when selecting ‘directly’ using the `DB` facade, and it happens before any Carbon conversion is attempted.

When fetched, the date is formatted as US-style date (short month name first, then date-of-month), with a 12-hour midnight timestamp (`12:00:00 AM`). In the timestamp, however, the space between the seconds and the meridian indicator is replaced with a colon, so that the output is `12:00:00:AM`.

This output is not a valid date format, so any attempts to convert it to a Carbon date fails with a `Carbon\Exceptions\InvalidFormatException`.

The error does not occur when fetching using a regular `PDO` object in PHP.

Columns of type `datetime` (or `datetime2`) and `datetimeoffset` are not affected.

### Steps To Reproduce:

Create a table in an SQL Server database which has a column of type `date`; example structure and content:

```
id - int, not null, PK
t_date - date, not null
t_datetime - datetime, not null
t_offset - datetimeoffset(7), not null

+----+------------+---------------------+------------------------------------+
| id | t_date     | t_datetime          | t_offset                           |
|----|------------|---------------------|------------------------------------|
| 1  | 2020-01-05 | 2020-01-05 11:28:31 | 2020-01-05 11:28:31.5870000 +00:00 |
+----+------------+---------------------+------------------------------------+
```

Select the column from the table (alone or along with other columns, makes no difference) using one of Laravel’s functions for fetching data from a table. Example using Tinker:

```
>>> DB::table('test_table')->first()

=> {#4961
     +"id": 1,
     +"t_date": "Jan  5 2022 12:00:00:AM", // <-- invalid date format!
     +"t_datetime": "2022-03-31 00:00:00", // <-- valid date format
     +"t_offset": "Aug 16 2018 01:29:57:5870000PM", // <-- valid date format
   }
```

Note how the `datetimeoffset` column is also converted to US style, but this time still a _valid_ date string which can be converted to a Carbon date.

**Comment by kokoshneta** [comment]
Oh bugger. I thought I’d looked closely, but apparently not – this has been reported before in #28491.

That issue was closed with no solution, though. It does adduce that this isn’t a new issue, but also not a universal one.

**Comment by driesvints** [comment]
There's been quite a number of these kind of issues posted over the years with SQL Server. I suggest you dig through the closed issue list a bit more to find some answers. They were always closed as invalid afaik.

**Comment by kokoshneta** [comment]
@driesvints I did look through quite a lot of SQL Server issues, and the only one I could find that matches this is the one I linked to above. All the others were about `datetime` columns – which work just fine for me – and were about different issues, like incorrect precision.

Could you point me to any other issues about this that have been posted? And the reason for their closure? #28491 was closed by the asker in order to investigate further, and then never revisited.

**Comment by driesvints** [comment]
It's best that you try a support channel:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

**Comment by LegitDongo** [comment]
Came across this when I was upgrading my homestead box and was getting the same

My issue was that my system was using fallback drivers(ODBC I think?) instead of the PECL drivers that you're supposed to use for SQLSRV connections

I had the install in my after.sh and forgot that it failed :)

**Comment by kokoshneta** [comment]
> My issue was that my system was using fallback drivers(ODBC I think?) instead of the PECL drivers that you're supposed to use for SQLSRV connections

This turned out to be the issue for me as well. It randomly vanished at some point last year after asking here, but that was because I installed the SQLSRV extensions for some other reason (to do with latency, if memory serves).

Updated to PHP 8.2, and the issue came back. After installing [version 5.11.0+ of the extensions](https://github.com/microsoft/msphpsql/releases/tag/v5.11.0), which support PHP 8.2, the issue immediately disappeared again (I had to [add these flags](https://github.com/microsoft/msphpsql/issues/651#issuecomment-881575586) for the installation to succeed).
